### ISSUE TITLE
Negate .prettierignore pattern

### ISSUE THREAD

**Issue by donaldpipowitch** [prettier/prettier#3328]
I'm not sure if this is a feature request or if I just can't find the docs. I'd like to negate a pattern inside the `.prettierignore` similar to `.gitignore`. Instead of blacklisting files to ignore I'd like to whitelist files to format. So I basically want to ignore everything except certain folder. Something like this:

```
*
!src/
!tests/
```

It looks like this is possible currently. Did  I get the syntax wrong?

**Comment by azz** [comment]
What you posted works for me. 

Probably better to do `prettier --write "{src/tests}/**/*.js"` though, might be a bit faster.

**Comment by goliney** [comment]
It doesn't work for me either. 

```
# .prettierignore
*
!client/js/**/*.js
!tests/js/**/*.js
```

When I run `prettier --write client/js/component.js` nothing changes, but when I clear `.prettierignore` and run the same command I see that the file is reformatted.

Prettier version: 1.9.1

**Comment by duailibe** [comment]
TIL: negate with `/*` works

That was totally unclear to me but this is how is described in [`.gitignore` docs](https://git-scm.com/docs/gitignore):

> Example to exclude everything except a specific directory `foo/bar` (note the `/*` - without the slash, the wildcard would also exclude everything within `foo/bar`):

    $ cat .gitignore
    # exclude everything except directory foo/bar
    /*
    !/foo
    /foo/*
    !/foo/bar

**Comment by goliney** [comment]
@duailibe thanks for the hint! Much appreciated!

This is what I've come up to:
```
# exclude everything
/*

# except /client/js/**/*.js
!/client/
/client/*
!/client/js
/client/js/**/*.*
!/client/js/**/*.js

# except /tests/js/**/*.js
!/tests/
/tests/*
!/tests/js
/tests/js/**/*.*
!/tests/js/**/*.js
```

**Comment by duailibe** [comment]
@goliney Thanks for confirming it works!

**Comment by donaldpipowitch** [comment]
Sorry, it just doesn't seem to work for me. If I understand you correctly I should switch to this:

```
/*
!/src/
!/tests/
```

But `src/index.ts` will still be ignored. (Tested with `prettier-vscode` which respects `.prettierignore`.)

**Comment by lydell** [comment]
@donaldpipowitch I think you have to exclude all _files_ within src and tests as well, like in @goliney’s example.

**Comment by CiGit** [comment]
@donaldpipowitch prettier-vscode@1.0.0 currently has a bug with ignore. Your `.prettierignore` seems to work as expected on master (not yet released) patch: prettier/prettier-vscode#303

**Comment by donaldpipowitch** [comment]
Thank you both. I'll try it again soon with the updated `prettier-vscode`.
